### PR TITLE
[train] fix fashion mnist example

### PR DESCRIPTION
**Pull request by matthewdeng**
## Why are these changes needed?

Making some minor fixes.

1. Update input `batch_size` to be global batch size. Introduce `worker_batch_size` so each iteration trains same global batch size.
2. Update dataset `size` calculation to only refer to the fraction of the data that is trained on each worker. This allows calculations (e.g. training progress, accuracy) to be correct.
3. Add `model.train()` for generality.
4. Remove `smoke-test` flag since it's not really being used.

## Related issue number

https://discuss.ray.io/t/train-fashion-mnist-example-accuracy-drops-when-num-workers-1/4735

## Checks

Successfully ran `python train_fashion_mnist_example.py --smoke-test --num-workers 4`

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by amogkam — APPROVED** [review]
Thanks for the quick fix!
